src/GraphQL/Types/QueryArgument.cs

```diff
@@ -37,8 +40,35 @@ public QueryArgument(Type type)
 
         public object DefaultValue { get; set; }
 
-        public IGraphType ResolvedType { get; set; }
+        public IGraphType ResolvedType
+        {
+            get => _resolvedType;
+            set => _resolvedType = CheckResolvedType(value);
+        }
+
+        public Type Type
+        {
+            get => _type;
+            private set => _type = CheckType(value);
+        }
+
+        private Type CheckType(Type type)
+        {
+            if (type?.IsInputType() == false)
+                throw Create(nameof(Type), type);
+
+            return type;
+        }
+
+        private IGraphType CheckResolvedType(IGraphType type)
+        {
+            if (!(type.GetNamedType() is GraphQLTypeReference) && type?.IsInputType() == false)
```

Comment:
Just noticed, I think missing a `type?` here on the `type.GetNamedType()`.

Maybe add a test with null for the `ResolvedType` property?

Reply:
I already raised a finger over the merge button :)

```c#
  public static IGraphType GetNamedType(this IGraphType type)
        {
            return type switch
            {
                NonNullGraphType nonNull => GetNamedType(nonNull.ResolvedType),
                ListGraphType list => GetNamedType(list.ResolvedType),
                _ => type
            };
        }
```

Reply:
I would say that on the contrary we can remove `?` when used before `IsInputType` or `IsOutputType` (`IsOutputType` requires small fix).

Reply:
![image](https://user-images.githubusercontent.com/21261007/71383664-5fec7e80-25ee-11ea-9fca-2a34f99a2c3c.png)

Nope, nothing to fix.

Reply:
I would still like to see a test that sets `ResolvedType` to null to confirm the behavior.

Reply:
Ok. I was wrong about `IsInputType`, `bool?` vs `bool`.

Reply:
Done